h2/stream.py

```diff
@@ -1020,17 +1031,42 @@ def stream_reset(self, frame):
 
         return [], events
 
+    def _build_hdr_validation_flags(self, events):
+        """
+        Constructs a set of header validation flags for use when normalizing
+        and validating header blocks.
+        """
+        try:
+            is_trailer = isinstance(
+                events[0], (_TrailersSent, TrailersReceived)
+            )
+        except IndexError:
+            is_trailer = False
```

Comment:
Aha, ok. This feels like an easy source of bugs, but for now let's put a comment here.